ai_ta_backend/service/retrieval_service.py

```diff
@@ -218,13 +220,28 @@ def llm_monitor_message(self, messages: List[str], course_name: str) -> List[Dic
       except:
         message_content = message['content']
 
+        # use a unique identifier for each message to check if it was already seen
+        conversation_id = message.get('conversation_id', '')
```

Comment:
I'm pretty sure the message does NOT contain `conversation_id` so this will be an emptry string every time, so the hashing won't work. Probably best to remove this for now, focus on a database-based solution.